File: cli/template/packages/scala/src/test/scala/do_/{{name}}/{{Name}}ClientTest.scala
package do_.{{name}}

import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers

class {{Name}}ClientTest extends AnyFlatSpec with Matchers {

  "{{Name}}Client" should "create with default options" in {
    val client = {{Name}}Client()
    client.baseUrl shouldBe "https://{{name}}.do"
  }

  it should "create with custom base URL" in {
    val client = {{Name}}Client(baseUrl = "https://custom.{{name}}.do")
    client.baseUrl shouldBe "https://custom.{{name}}.do"
  }

  it should "accept API key" in {
    val client = {{Name}}Client(apiKey = Some("test-key"))
    client shouldBe a[{{Name}}Client]
  }

  it should "create with options case class" in {
    val options = {{Name}}ClientOptions(
      apiKey = Some("test-key"),
      baseUrl = "https://test.{{name}}.do"
    )
    val client = new {{Name}}Client(options)
    client.baseUrl shouldBe "https://test.{{name}}.do"
  }
}

File: cli/template/packages/kotlin/src/test/kotlin/do_/{{name}}/{{Name}}ClientTest.kt
package do_.{{name}}

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class {{Name}}ClientTest {

    @Test
    fun `should create client with api key`() {
        val client = {{Name}}Client("test-key")
        assertEquals("test-key", client.apiKey)
        assertEquals("https://{{name}}.do", client.baseUrl)
    }

    @Test
    fun `should create client with custom base url`() {
        val client = {{Name}}Client("test-key", "https://custom.{{name}}.do")
        assertEquals("https://custom.{{name}}.do", client.baseUrl)
    }

    @Test
    fun `should create client without api key`() {
        val client = {{Name}}Client()
        assertNull(client.apiKey)
    }
}
